src/Microsoft.ML.Data/Transforms/FeatureContributionCalculationTransform.cs

```diff
@@ -0,0 +1,367 @@
+            }
+
+            /// <summary>
+            /// Returns the input columns needed for the requested output columns.
+            /// </summary>
+            private protected override Func<int, bool> GetDependenciesCore(Func<int, bool> activeOutput)
+            {
+                var active = new bool[InputSchema.Count];
+                InputSchema.TryGetColumnIndex(_parent.FeatureColumn, out int featureCol);
+                active[featureCol] = true;
+                return col => active[col];
+            }
+
+            public override void Save(ModelSaveContext ctx)
+                => _parent.Save(ctx);
+
+            // The FeatureContributionCalculatingTransformer produces two sets of columns: the columns obtained from scoring and the FeatureContribution column.
+            // If the argument stringify is true, the type of the FeatureContribution column is string, otherwise it is a vector of float.
+            protected override Schema.DetachedColumn[] GetOutputColumnsCore()
+            {
+                // Add FeatureContributions column.
+                var builder = new MetadataBuilder();
+                builder.Add(InputSchema[_featureColumnIndex].Metadata, x => x == MetadataUtils.Kinds.SlotNames);
+                return new[] { new Schema.DetachedColumn(DefaultColumnNames.FeatureContributions, new VectorType(NumberType.R4, _featureColumnType.ValueCount), builder.GetMetadata()) };
+            }
+
+            protected override Delegate MakeGetter(Row input, int iinfo, Func<int, bool> active, out Action disposer)
+            {
+                disposer = null;
+                if (active(iinfo))
+                    return GetContributionGetter(input, _featureColumnIndex);
+                return null;
+            }
+
+            public Delegate GetContributionGetter(Row input, int colSrc)
+            {
+                Contracts.CheckValue(input, nameof(input));
+                Contracts.Check(0 <= colSrc && colSrc < input.Schema.Count);
+
+                var typeSrc = input.Schema[colSrc].Type;
+                Func<Row, int, ValueGetter<VBuffer<float>>> del = GetValueGetter<int>;
+
+                // REVIEW: Assuming Feature contributions will be VBuffer<float>.
+                // For multiclass LR it needs to be(VBuffer<float>[].
+                var meth = del.GetMethodInfo().GetGenericMethodDefinition().MakeGenericMethod(typeSrc.RawType);
+                return (Delegate)meth.Invoke(this, new object[] { input, colSrc });
+            }
+
+            private ValueGetter<VBuffer<float>> GetValueGetter<TSrc>(Row input, int colSrc)
+            {
+                Contracts.AssertValue(input);
+                Contracts.AssertValue(_parent._predictor);
+
+                var featureGetter = input.GetGetter<TSrc>(colSrc);
+
+                // REVIEW: Scorer can do call to Sparicification\Norm routine.
```

Comment:
>Sparicification [](start = 49, length = 15)

My Inglish is far unferior, so I don't know this word. Maybe 'sparsification'?

Reply:
also `can do call` sounds a bit alien

---
In reply to: [243096303](https://github.com/dotnet/machinelearning/pull/1847#discussion_r243096303) [](ancestors = 243096303)

Reply:
Yes, It was from the getter in the other file, will update thank you Pete

---
In reply to: [243096400](https://github.com/dotnet/machinelearning/pull/1847#discussion_r243096400) [](ancestors = 243096400,243096303)